lib/handler/redirect.js

```diff
@@ -92,7 +92,10 @@ class RedirectHandler {
         See comment on onData method above for more detailed informations.
       */
 
-      this.agent.dispatch(this.opts, this.handler)
+      this.location = null
+      this.abort = null
+
+      this.agent.dispatch(this.opts, this)
```

Comment:
@dnlup here it's reused. Currently it would create a new RedirectHandler in Agent.dispatch.

Reply:
Ahh, I missed it. Thanks.